### PR TITLE
Compact job autodetects previous date

### PR DESCRIPTION
**Pull request by lyle-adeptmind**

### CODE REVIEW

File: summary/__main__.py [hunk 1]
```diff
@@ -161,7 +161,7 @@ def main():
     indexes = config.INDEXES_SPEC
     collections = config.COLLECTIONS_SPEC
     for name in collections:
-        collections[name] = collections[name] + "-" + datetime.now().strftime("%Y-%m-%d")
+        collections[name] = collections[name] + datetime.now().strftime("-%Y-%m-%d")
```
Comment (SiweiYang) [inline summary/__main__.py:164]: i'd say either is cool

File: summary/db/mongoclient.py [hunk 2]
```diff
@@ -11,13 +11,20 @@ def __init__(self, host='localhost', port=27017, db=''):
     """
         TBD, implement when necessary
     """
+    def find_all(self, collection):
+        try:
+            tableref = self.client[self.db][collection]
+            return tableref.find({})
+        except Exception as ex:
+            logging.exception(ex)
```
Comment (SiweiYang) [inline summary/db/mongoclient.py:19]: by default we should escalate, unless explicitly suppressed

File: summary/db/mongoclient.py [hunk 3]
```diff
@@ -29,3 +36,12 @@ def write_many(self, collection, data):
...
             logging.exception(ex)
             return []
+
+    def list_collections(self, filter=None):
+        try:
+            dbref = self.client[self.db]
+            collection_names = dbref.list_collection_names(filter=filter, nameOnly=True)
+            return collection_names
+        except Exception as ex:
+            logging.exception(ex)
```
Comment (SiweiYang) [inline summary/db/mongoclient.py:46]: same here

File: summary/__main__.py (hunk 1 above)
Comment (lyle-adeptmind) [reply summary/__main__.py:164]: Yeah, this way is fewer string concats.

**Pull request merged**